### PR TITLE
[roku] Improve exception handling for failed http connections

### PR DESCRIPTION
**Pull request by mlobstein**
Improve handling of http exceptions. If the http GET request to the roku device fails, xml parsing will now be skipped.

### CODE REVIEW

File: bundles/org.openhab.binding.roku/src/main/java/org/openhab/binding/roku/internal/communication/RokuCommunicator.java
Comment (jlaur) [reply bundles/org.openhab.binding.roku/src/main/java/org/openhab/binding/roku/internal/communication/RokuCommunicator.java:288]: Also worth checking while refactoring this anyway: Restore the interrupted state of the thread by calling `Thread.currentThread().interrupt()`, see https://rules.sonarsource.com/java/RSPEC-2142/

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks for the fix! Only a few minor comments.

**Comment by mlobstein** [comment]
> Thanks for the fix! Only a few minor comments.

Comments have been addressed, Happy Thanksgiving!
I give thanks to you and all the maintainers for the innumerable reviews!

**Review by jlaur — APPROVED** [review]
LGTM. Will let @morph166955 have a look also.

**Comment by jlaur** [comment]
@morph166955 - would you like to have a look before merging?

**Comment by morph166955** [comment]
Sorry I've been out of town for the holiday here.  I haven't had a chance to sit and go through anything in about a week now.

**Comment by morph166955** [comment]
LGTM!

**Review by morph166955 — APPROVED** [review]

**Pull request merged**
